.clang-tidy

```diff
@@ -59,8 +58,6 @@ modernize-*,
 performance-*,
 portability-*,
 readability-*,
--readability-braces-around-statements,
```

Comment:
@TonyXiang8787 this relates to the `.clang-format` as well. do you have a strong opinion about not adding braces around statements? If not, I suggest we go with the LLVM convention of adding them. In that case, we should also update the `.clang-format` file with that (provided that the `diff` is small, which I expect to be the case after this PR)

Reply:
What braces are you referring?

Reply:
the braces in the case of short `if` statements, e.g.:

```cpp
if (foo)
    bar;
else
    baz;

etc
```

vs.

```cpp
if (foo) {
    bar;
}
else {
    baz;
}

etc
```

Reply:
My opinion is: always with braces when you put the statement in a new line. If you put the statement at the same line of `if`, you can omit braces.

Reply:
ok, then I will update the `.clang-format` file accordingly

Reply:
investigated this and it's probably not desired to add this to the `.clang-format` file since it doesn't know semantics and therefore could lead to incorrect formatting. instead, let's keep it in the `.clang-tidy` file (because that actually takes semantics into account).